frida_mode/Makefile

```diff
@@ -1,86 +1,58 @@
 PWD:=$(shell pwd)/
+ROOT:=$(shell realpath $(PWD)..)/
 INC_DIR:=$(PWD)include/
 SRC_DIR:=$(PWD)src/
 INCLUDES:=$(wildcard $(INC_DIR)*.h)
-SOURCES:=$(wildcard $(SRC_DIR)*.c)
+SOURCES:=$(wildcard $(SRC_DIR)**/*.c) $(wildcard $(SRC_DIR)*.c)
 BUILD_DIR:=$(PWD)build/
-CFLAGS+=-fPIC -D_GNU_SOURCE
+CFLAGS+=-fPIC -D_GNU_SOURCE -Wno-prio-ctor-dtor
+QEMU_INC_DIR:=$(ROOT)qemu_mode/qemuafl/qemuafl/
+QEMU_INC_API:=$(QEMU_INC_DIR)api.h
 
 FRIDA_BUILD_DIR:=$(BUILD_DIR)frida/
-FRIDA_TRACE:=$(FRIDA_BUILD_DIR)afl-frida-trace.so
+FRIDA_TRACE:=$(BUILD_DIR)afl-frida-trace.so
+FRIDA_TRACE_EMBEDDED:=$(BUILD_DIR)afl-frida-trace-embedded
 
 ARCH=$(shell uname -m)
 ifeq "$(ARCH)" "aarch64"
  ARCH:=arm64
- TESTINSTR_BASE:=0x0000aaaaaaaaa000
-endif
-
-ifeq "$(ARCH)" "x86_64"
- TESTINSTR_BASE:=0x0000555555554000
 endif
 
 ifeq "$(shell uname)" "Darwin"
  OS:=macos
- AFL_FRIDA_INST_RANGES=0x0000000000001000-0xFFFFFFFFFFFFFFFF
  CFLAGS:=$(CFLAGS) -Wno-deprecated-declarations
- TEST_LDFLAGS:=-undefined dynamic_lookup
 endif
+
 ifeq "$(shell uname)" "Linux"
  OS:=linux
- AFL_FRIDA_INST_RANGES=$(shell $(PWD)test/testinstr.py -f $(BUILD_DIR)testinstr -s .testinstr -b $(TESTINSTR_BASE))
- CFLAGS:=$(CFLAGS) -Wno-prio-ctor-dtor
- TEST_LDFLAGS:=
 endif
 
 ifndef OS
  $(error "Operating system unsupported")
 endif
 
-VERSION=14.2.13
-GUM_DEVKIT_FILENAME=frida-gum-devkit-$(VERSION)-$(OS)-$(ARCH).tar.xz
-GUM_DEVKIT_URL="https://github.com/frida/frida/releases/download/$(VERSION)/$(GUM_DEVKIT_FILENAME)"
+GUM_DEVKIT_VERSION=14.2.13
```

Comment:
current is 14.2.17
do you want to pin it to that specific version? or update the version from time to time?
you could also get the most current version automatically and if in the future this fails then its pinned by hand (or fixed)?

Reply:
I have bumped the version to the latest. I think I would rather hard-code the version so we have the opportunity to test new versions rather than potentially hit issues in the unlikely event that a new FRIDA release breaks something. Given the makefile is now very simple, I would expect someone wanting to use a newer version would be easily able to do so. In any case, I expect that people hacking on this would actually prefer to build FRIDA from source, but I didn't want to add it as submodules and incur all of their dependencies and additional compilation time.